### PR TITLE
dont crawl duplicate modules when loading plugins

### PR DESCRIPTION
**Pull request by habdelra**

### CODE REVIEW

File: packages/hub/plugin-loader.js [hunk 1]
```diff
@@ -114,22 +115,23 @@ class PluginLoader {
 
   async _crawlPlugins(dir, outputPlugins, seen, includeDevDependencies, breadcrumbs) {
     log.trace("plugin crawl dir=%s, includeDevDependencies=%s, breadcrumbs=%j", dir, includeDevDependencies, breadcrumbs);
-    if (seen[dir]) {
-      if (seen[dir].attributes && seen[dir].attributes.includedFrom) {
+    let realdir = await realpath(dir);
+    let packageJSON = path.join(realdir, 'package.json');
+    let json = require(packageJSON);
```
Comment (ef4) [inline packages/hub/plugin-loader.js]: I think this eliminated an optimization -- we're going to re-parse the package.json for directories we've already seen, where before we hit the early `return`and skipped that.

**Comment by habdelra** [comment]
make sense. will do

File: packages/hub/plugin-loader.js (hunk 1 above)
Comment (habdelra) [reply packages/hub/plugin-loader.js]: thanks, i'll fix that

**Comment by habdelra** [comment]
Actually I wonder if throwing will continue to make it just as brittle as it is now. Keep in mind that when we have minor misalignments in @cardstack/* versions, it can cause hundreds of duplicate modules to be loaded, which can cause healthcheck failires on start up. How about a warning message instead, so the hub can be resilient, but still be transparent that not everything is ok.

**Comment by ef4** [comment]
Ok, but lets make the warning an error when it happens inside tests.

**Comment by habdelra** [comment]
sure that's fine. to clarify, when you say "error" do you want to throw and error, or log at an error level?

**Comment by ef4** [comment]
Throw an error. Something that will make the tests fail.

**Comment by habdelra** [comment]
@ef4 when running our ember-tests in this repo, I'm running into dupe modules being loaded like this: ```Error: The plugin module name '@cardstack/codegen' has already been loaded from the module path /Users/hassan/codez/cardstack/packages/codegen, conflict with load of module at path /Users/hassan/codez/cardstack/node_modules/@cardstack/codegen.``` (and because its the test env, we blow up). How do you think I should handle these?

**Comment by ef4** [comment]
I think one of those paths is a symlink to the other? In that case we should run them through fs.realpath before comparing.

**Comment by habdelra** [comment]
cool--thanks! I'll give that a shot

**Pull request merged**